### PR TITLE
Revert #17398 and #17402

### PR DESCRIPTION
**Pull request by narimiran**
Another breaking change which shouldn't be part of Nim 1.6

### CODE REVIEW

**Comment by juancarlospaco** [comment]
If someone really relies on broken behaviours and not asserting its inputs,
then that person should do a tiny "shim", instead of keeping the bad code for everyone,
is very cheap to check if a `char` is `DirSep`. 🤷

```nim
template getHomeDirShim*(): string =
  let s = getHomeDir()
  if s[^1] == '/': s else: s & '/'
```

**Comment by timotheecour** [comment]
or, using `getHomeDir().normalizePathEnd(trailingSep = false)`

**Comment by kaushalmodi** [comment]
Wouldn't it be better to create a version-1p6 branch and undo all the breaking commits there? I thought devel was targeting the latest and greatest future Nim version.

**Comment by Araq** [comment]
> Wouldn't it be better to create a version-1p6 branch and undo all the breaking commits there? I thought devel was targeting the latest and greatest future Nim version.

We're considering it.

**Comment by timotheecour** [comment]
> Wouldn't it be better to create a version-1p6 branch and undo all the breaking commits there? I thought devel was targeting the latest and greatest future Nim version.

this is would be much better than reverting these commit, likewise with other reverts that are currently pending and other similar reverts that were made in 1.5; but we don't even need to undo those commits in that branch and in fact it's better not to as it allows gradual migration, see https://github.com/nim-lang/Nim/pull/18486

File: lib/pure/os.nim [hunk 1]
```diff
@@ -901,14 +901,9 @@ proc getHomeDir*(): string {.rtl, extern: "nos$1",
...
-    # `getHomeDir()` doesn't end in `DirSep` even if `$HOME` (on posix) or
-    # `$USERPROFILE` (on windows) does, unless `-d:nimLegacyHomeDir` is specified.
-    from std/strutils import endsWith
-    assert not getHomeDir().endsWith DirSep
 
-  when defined(windows): result = getEnv("USERPROFILE")
-  else: result = getEnv("HOME")
-  result.normalizePathEnd(trailingSep = defined(nimLegacyHomeDir))
+  when defined(windows): return getEnv("USERPROFILE") & "\\"
+  else: return getEnv("HOME") & "/"
```
Comment (Araq) [inline lib/pure/os.nim:906]: So previously the code used a trailing dir **on purpose**. Calling this a "bugfix" then is a bit of stretch...

**Comment by Araq** [comment]
> then that person should do a tiny "shim", instead of keeping the bad code for everyone

And why is it suddenly "bad code"? It used to work and could continue to work...

**Comment by timotheecour** [comment]
See https://github.com/nim-lang/Nim/pull/18513 (`{.migrated:... .}`) which provides a much better solution, by issuing a new migration warning for these 3 APIs, which allows you to find at CT all the callsites that need to be checked; the migration warning can be turned off on a per-API basis (eg `-d:nimMigratedGetHomeDir`) which should make the migration a lot easier.

> And why is it suddenly "bad code"? It used to work and could continue to work...

because it was inconsistent both with all other languages and all other APIs in nim that return a dir, see long list here: https://github.com/nim-lang/Nim/issues/17393#issuecomment-881610159, consistency is still a thing.

**Comment by Araq** [comment]
> because it was inconsistent both with all other languages and all other APIs in nim that return a dir, see long list here: #17393 (comment), consistency is still a thing.

But now it's not consistent with its own previous behavior anymore... Great consistency here.

**Pull request merged**